### PR TITLE
rework std.atomic

### PR DESCRIPTION
**Pull request by andrewrk**
* remove std.atomic.Queue
* remove std.atomic.Stack
* remove `std.atomic.Ordering`
  - it is provided by the language with `std.builtin.AtomicOrder`.
* remove `std.atomic.fence`
  - it is provided by the language with `@fence`.
* remove `std.atomic.compilerFence`
  - if this is desired, it should be a language feature, not a standard
    library function with inline asm.
* move std.atomic.Atomic to std.atomic.Value
* fix incorrect argument order passed to testing.expectEqual
* make the functions be a thin wrapper over the atomic builtins and
  stick to the naming conventions.
* remove pointless functions loadUnchecked and storeUnchecked. Instead,
  name the field `raw` instead of `value` (which is redundant with the
  type name).
* simplify the tests by not passing every possible combination. Many
  cases were iterating over every possible combinations but then not
  even using the for loop element value!
* remove the redundant compile errors which are already implemented by
  the language itself.
* remove dead x86 inline assembly. this should be implemented in the
  language if at all.

### CODE REVIEW

**Comment by xdBronch** [comment]
why remove Queue and Stack?

**Comment by andrewrk** [comment]
Have a look at the respective commit messages

**Comment by xdBronch** [comment]
sorry, didnt think to check that

File: lib/std/atomic.zig [hunk 1]
```diff
@@ -1,40 +1,374 @@
...
+
+        pub inline fn store(self: *Self, value: T, comptime order: AtomicOrder) void {
+            @atomicStore(T, &self.raw, value, order);
+        }
+
+        pub inline fn swap(self: *Self, operand: T, comptime order: AtomicOrder) T {
+            return @atomicRmw(T, &self.raw, .Xchg, operand, order);
+        }
+
+        pub inline fn cmpxchgWeak(
```
Comment (kprotty) [inline lib/std/atomic.zig:51]: Given this was renamed to the builtin, why not rename `swap` as well?

**Comment by andrewrk** [comment]
yeah I know. the event loop needs to be completely rewritten.

**Pull request merged**